std/datetime/systime.d

```diff
@@ -9031,22 +9069,64 @@ private:
       +/
     @property void adjTime(long adjTime) @safe nothrow scope
     {
-        _stdTime = _timezone.tzToUTC(adjTime);
+        _stdTime = timezoneHack.tzToUTC(adjTime);
     }
 
 
-    // Commented out due to bug http://d.puremagic.com/issues/show_bug.cgi?id=5058
-    /+
-    invariant()
+    final class InitTimeZone : TimeZone
+    {
+    public:
+
+        static immutable(InitTimeZone) opCall() @safe pure nothrow @nogc { return _initTimeZone; }
+
+        @property override bool hasDST() @safe const nothrow @nogc { return false; }
+
+        override bool dstInEffect(long stdTime) @safe const nothrow @nogc { return false; }
+
+        override long utcToTZ(long stdTime) @safe const nothrow @nogc { return 0; }
+
+        override long tzToUTC(long adjTime) @safe const nothrow @nogc { return 0; }
+
+        override Duration utcOffsetAt(long stdTime) @safe const nothrow @nogc { return Duration.zero; }
+
+    private:
+
+        this() @safe immutable pure
+        {
+            super("SysTime.init's timezone", "SysTime.init's timezone", "SysTime.init's timezone");
+        }
+
+        static immutable InitTimeZone _initTimeZone = new immutable(InitTimeZone);
+    }
+
+    // Assigning a value to _timezone in SysTime.init currently doesn't work due
+    // to https://issues.dlang.org/show_bug.cgi?id=17740. So, to hack around
+    // that problem, this accessor has been added to check for null and use
+    // InitTimeZone for SysTime.init (which is the only case where _timezone
+    // would be null). This thus fixes the problem with segfaulting when using
+    // SysTime.init but at the cost of what should be an unnecessary null check.
+    // Once 17740 has finally been fixed, all uses of timezoneHack should be
+    // replaced with _timezone again, and _timezone should be directly assigned
+    // InitTimeZone().
+    pragma(inline, true) @property timezoneHack() @safe const pure nothrow @nogc
+    {
+        return _timezone is null ? InitTimeZone() : _timezone;
+    }
+
+    // https://issues.dlang.org/show_bug.cgi?id=17732
+    @safe unittest
     {
-        assert(_timezone !is null, "Invariant Failure: timezone is null. Were you foolish enough to use " ~
-                                   "SysTime.init? (since timezone for SysTime.init can't be set at compile time).");
+        assert(SysTime.init.timezone is InitTimeZone());
+        assert(SysTime.init.toISOString() == "00010101T000000+00:00");
+        assert(SysTime.init.toISOExtString() == "0001-01-01T00:00:00+00:00");
+        assert(SysTime.init.toSimpleString() == "0001-Jan-01 00:00:00+00:00");
+        assert(SysTime.init.toString() == "0001-Jan-01 00:00:00+00:00");
     }
-    +/
 
 
     long  _stdTime;
     Rebindable!(immutable TimeZone) _timezone;
+    //Rebindable!(immutable TimeZone) _timezone = InitTimeZone();
```

Comment:
What is this?

Reply:
As the comment above states, this is what `_timezone` is supposed to be set to but can't currently be set to until the compiler bug is fixed.